smoke/features/steps/steps.py

```diff
@@ -324,26 +324,29 @@ def verifyRoute(context):
 @then(u'Trigger the build using oc start-build openshift-jee-sample')
 def startBuild(context):
     triggerbuild('openshift-jee-sample',current_project)
-    time.sleep(300)
-
 
 @then(u'verify the build status of openshift-jee-sample-docker build is Complete')
 def verifyDockerBuildStatus(context):
-    buildState = oc.get_resource_info_by_jsonpath('build','openshift-jee-sample-docker-1',current_project,json_path='{.status.phase}')
-    if not 'Complete' in buildState:
-        raise AssertionError
-    else:
-        print("Build openshift-jee-sample-docker-1 status:{buildState}")
-    
+    verify_status('build', 'openshift-jee-sample-docker-1', 2, 10, 'Complete', '{.status.phase}')
+
 
 @then(u'verify the build status of openshift-jee-sample-1 is Complete')
 def verifyJenkinsBuildStatus(context):
-    buildState = oc.get_resource_info_by_jsonpath('build','openshift-jee-sample-1',current_project,json_path='{.status.phase}')
-    if not 'Complete' in buildState:
-        raise AssertionError
-    else:
-        print("Build openshift-jee-sample-1-deploy status:{buildState}")
+    verify_status('build', 'openshift-jee-sample-1', 2, 100, 'Complete', '{.status.phase}')
+
 
+def verify_status(object_type, object_name, sleep_delay, sleep_count, expected_status,json_path='{.status.phase}'):
```

Comment:
This function effectively polls the object's status to check a specific condition. I'd rename the parameters to be a bit clearer:

```suggestion
def verify_status(object_type, object_name, poll_interval_seconds, max_retries, expected_status,json_path='{.status.phase}'):
```